### ISSUE TITLE
HTTPRoute: Endless blocking issue due to absence of logic for receiving from channels

### ISSUE THREAD

**Issue by parkjeongryul** [Kong/kubernetes-ingress-controller#4260]
### Is there an existing issue for this?

- [X] I have searched the existing issues

I've already reported this issue in kong repo, but it doesn't seem to be getting tracked. 
https://github.com/Kong/kong/issues/10923

I might have reported it in the wrong repo, so I've created it here as well

### Current Behavior

PATCH commit: https://github.com/parkjeongryul/kubernetes-ingress-controller/commit/e0d7145dc99d93810f7d738482a05af83e68137b

There is `kubernetesObjectStatusQueue`. (which size is 8192)

But there is only a PUBLISHER(send to chan) for that queue and no SUBSCRIBERS(receieve from chan)... 
(And why do we need this queue?) 

So it seems that after a certain GKV's 8192 event (creation or deletion), the kong ingress controller is forever broken.

### Expected Behavior

The reconcile function of HTTPRoute must always operate correctly.

### Steps To Reproduce

```markdown
This is our destroy-kong.sh
The behavior is reproduced near the end of the script.


user@AL01974378 vks % cat destroy-kong.sh 
#!/bin/bash

# Function to create and delete HttpRoute
create_and_delete_httproute() {
  for j in {1..100}
  do
    # Define HttpRoute, Service, and Deployment YAML content as a bash variable.
    RESOURCES_YAML=$(cat <<EOF
apiVersion: v1
kind: Namespace
metadata:
  name: echo-ns-$1
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: echo-tmp-$1
  namespace: echo-ns-$1
  annotations:
    konghq.com/strip-path: 'true'
spec:
  parentRefs:
  - name: clous-api
    namespace: clous-api
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /echo-tmp-$1
    backendRefs:
    - name: echo
      kind: Service
      port: 1027
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: echo
  name: echo
  namespace: echo-ns-$1
spec:
  ports:
  - port: 1025
    name: tcp
    protocol: TCP
    targetPort: 1025
  - port: 1026
    name: udp
    protocol: TCP
    targetPort: 1026
  - port: 1027
    name: http
    protocol: TCP
    targetPort: 1027
  selector:
    app: echo
EOF
    )

    # Create HttpRoute, Service, and Deployment
    echo "Creating resources in echo-ns-$1... $j steps"
    echo "${RESOURCES_YAML}" | kubectl apply -f -

    # Delete Namespace
    echo "Deleting namespace echo-ns-$1... $j steps"
    kubectl delete ns echo-ns-$1
  done
}

# Start 40 parallel tasks to create and delete HttpRoutes, Services, and Deployments
for i in {1..40}
do
  create_and_delete_httproute $i &
done

# Wait for all
wait
```

### Kong Ingress Controller version

```shell
v2.9.2
```

### Kubernetes version

```shell
v1.23.15
```

### Anything else?

**Comment by czeslavo** [comment]
The [fix](https://github.com/Kong/kubernetes-ingress-controller/pull/4267) for this was merged to `main`, it will be backported to `release/2.10.x` and released as part of `2.10.2`.

**Comment by parkjeongryul** [comment]
Awesome 👍
I'm gonna test with latest version.
